src/port/port.tcl

```diff
@@ -2734,6 +2734,17 @@ proc action_selfupdate { action portlist opts } {
     }
 
     if {$base_updated} {
+        for {set i 0} {$i < $::argc} {incr i} {
+            if {[lindex $::argv $i] == "selfupdate"} {
```

Comment:
This will turn a invocation of `sudo port selfupdate --nosync` into `sudo port sync --nosync`, which will fail because it does not recognize the flag. I see multiple options to fix this:

1. Drop all parameters after `selfupdate` and run `sync`.
2. Just re-run the same command again. This will run `selfupdate` twice, but avoid having to put the "you should run port upgrade outdated" message above the sync step.
3. Find out whether `--nosync` was passed and exit immediately instead of running `sync` if it was.

Also note that MacPorts does have an interactive mode which can be started using `sudo port`. If the interactive mode was started, there will be no `selfupdate` in `$::argv` and your replacement will fail. You'd still re-exec but the ports tree wouldn't have been updated. Since we can't tell MacPorts to 'run sync, then start interactive mode' and we've previously exited interactive mode after a selfupdate, I'd say it's fine to just run `port sync` (or, as discussed above, selfupdate again) and then exit.

Reply:
Hmm…given your concerns, I think the second option is probably the safest and least likely to break down the line if the set of options that `selfupdate` and `sync` take diverge more. I think I'll tweak the message in this case to explain that we're running `selfupdate` again: this at least gives the user a bit of context as to what we're doing, and should only happen infrequently anyways so I don't think it's a huge issue.

Reply:
@neverpanic It's been a while since you looked at this last but would you mind doing another pass?